isis/src/qisis/objs/JigsawSetupDialog/JigsawSetupDialog.cpp

```diff
@@ -99,10 +101,12 @@ namespace Isis {
     if (!m_project->directory()->model()->selectedItems().isEmpty()) {
       foreach (ProjectItem * projItem, m_project->directory()->model()->selectedItems()) {
         if (projItem->isImage()) {
+          projItem->setData(QVariant(QColor(Qt::transparent)), Qt::UserRole+10);
```

Comment:
**_(this isn't something you need to do, i just thought of this as an idea that could possibly generalize our UserRoles)._**

Something we could do regarding the _Qt::UserRole+10_ is moving it to a public enum inside of ProjectItemModel.

```c++
// In ProjectItemModel header
enum UserRole {
  BundleObservationSolveSettingsColorRole = Qt::UserRole+2;
};
```

Then, in in JigsawSetupDialog.cpp and SortFlterProxyModel.cpp, we could replace the Qt::UserRole+1 with ProjectItemModel::BundleObservationSolveSettingsColorRole (which is long, the name could be anything meaningful for that role).

Alternatively, we could add the enum to a new header, like UserRoles.h, and put it in an UserRole namespace under the Isis namespace; or a globals header (Isis.h), but I'm not sure what the implications of that would be.

I believe that ProjectItemModel/ProjectItem already implicitly defines a data role, Qt::UserRole+1, in as a default parameter for ProjectItemModel::findItemData(..., role=Qt::UserRole+1) and ProjectItem::findItemData(..., role=Qt::UserRole+1). We could add a _DataRole = Qt::UserRole+1_ to the ProjectItemModel UserRole enum and replace the default role parameter values for findItemData with ProjectItemModel::DataRole;